doppler/resource_secret.go

```diff
@@ -85,7 +85,7 @@ func resourceSecretRead(ctx context.Context, d *schema.ResourceData, m interface
 
 	secret, err := client.GetSecret(ctx, project, config, name)
 	if err != nil {
-		return diag.FromErr(err)
+		return handleNotFoundError(err, d)
```

Comment:
If a secret doesn't exist, I don't think we return an error today. I think we just return null values for the raw and computed:

```
GET https://api.doppler.com/v3/configs/config/secret?project=aa&config=dev&name=MISSING_SECRET
#+END
HTTP/1.1 200 OK
date: Wed, 05 Apr 2023 13:06:31 GMT
...

#+RESPONSE
{
  "name": "MISSING_SECRET",
  "value": {
    "raw": null,
    "computed": null
  },
  "success": true
}
#+END
```

Reply:
Is there a problem having the handling in place for it anyways? This would future proof us if we ever change that approach in the future.

Reply:
We would want some kind of check here to be sure that the error is indeed a Not Found error but the problem is that we don't actually return errors when a secret was Not Found. If we want to keep the error handling, we'll probably need to parse the `SecretValue` with pointers (so they can be `nil`) and use `nil` as the Not Found indicator.

```diff
diff --git a/doppler/models.go b/doppler/models.go
index bf4faac..a4c4def 100644
--- a/doppler/models.go
+++ b/doppler/models.go
@@ -44,8 +44,8 @@ type Secret struct {
 }
 
 type SecretValue struct {
-	Raw      string `json:"raw"`
-	Computed string `json:"computed"`
+	Raw      *string `json:"raw"`
+	Computed *string `json:"computed"`
 }
 
 func getSecretId(project string, config string, name string) string {

```

This might have some other implications though; we'd need to test

Reply:
Is this check that it's already doing not enough?

```
if strings.Contains(err.Error(), "Could not find requested") {
  ...
}
```

Reply:
No, the `GET /v3/config/configs/secret` endpoint won't throw an error at all if a secret is missing, it just returns `null` values for the raw and comuted

Reply:
I just tested and with the error handling it still behaves the same way it did prior to the change if you delete a secret:

```
Terraform will perform the following actions:

  # doppler_secret.elasticsearch will be updated in-place
  ~ resource "doppler_secret" "elasticsearch" {
      + computed = (sensitive value)
        id       = "rocket.dev.ENDPOINT"
        name     = "ENDPOINT"
      + value    = (sensitive value)
        # (2 unchanged attributes hidden)
    }

Plan: 0 to add, 1 to change, 0 to destroy.
```

So, this change is essentially a no-op unless we ever change things so requesting a specific secret returns a not found error that matches what we're looking for.

Reply:
Right, because the API never throws an error, correct? I think we should remove the error handling there if we can't hit a Not Found error for that endpoint, otherwise it could create confusion about the expected behavior.

However, if you wanted to check for `nil` values and throw in that scenario, I think that would make sense. If you did that, it should behave as it does today, just with the extra warning.

Reply:
Ah, good call. In that case, I think it makes sense to keep the error handling